src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ObjectFormatting/Formatters/DebuggerDisplay/DebuggerDisplayParser.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay
+{
+    internal static class DebuggerDisplayParser
+    {
+        internal record struct ParsedDebuggerDisplay(string FormatString, List<Expression> Expressions);
+        internal record struct Expression(ReadOnlyMemory<char> ExpressionString, FormatSpecifier FormatSpecifier);
+
+        internal static ParsedDebuggerDisplay? ParseDebuggerDisplay(string debuggerDisplay)
+        {
+            //
+            // A debugger display value is a string with expressions inside that can be evaluated.
+            // This method will transform the debugger display into a standard format string
+            // and extract the expressions.
+            //
+            // Simplified example:
+            // Input: "Count = {Count}, Info = {DebuggerInfo()}"
+            // Output: FormatString="Count = {0}, Info = {1}", Expressions=["Count", "DebuggerInfo()"]
+            //
+            StringBuilder fmtString = new();
+            List<Expression> expressions = new();
+
+            for (int i = 0; i < debuggerDisplay.Length; i++)
+            {
+                char c = debuggerDisplay[i];
+                switch (c)
+                {
+                    case '\\':
+                        // Escape sequence is not currently supported.
+                        return null;
```

Comment:
Rather than returning null in cases that are malformed or not supported, could we actually return a short error message /code as the format string itself?  This would make it a lot easier to understand when making syntax errors.